src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelRepository.cs

```diff
@@ -341,12 +387,20 @@ public async Task<Guid> GetDirectChannelForMembersAsync(Guid member1Id, Guid mem
         {
             using (var connection = _sqlConnectionFactory.CreateConnection())
             {
-                var sqlQuery = @"SELECT c.Id
-                                    FROM Channels c
-                                    INNER JOIN ChannelMembers cm ON c.Id = cm.ChannelId
-                                    WHERE ((cm.MemberId = @member1Id OR cm.MemberId = @member2Id) AND (c.CreatorId = @member1Id OR c.CreatorId = @member2Id)) AND c.Type = 2
-                                    GROUP BY c.Id
-                                    HAVING COUNT(cm.MemberId) = 2";
+                var sqlQuery = $@"
+                    SELECT 
+                        c.{nameof(Channel.Id)}
+                    FROM 
+                        Channels c
+                    INNER JOIN ChannelMembers cm 
+                        ON c.{nameof(Channel.Id)} = cm.{nameof(ChannelMember.ChannelId)}
+                    WHERE
+                        (cm.{nameof(ChannelMember.MemberId)} = @{nameof(member1Id)} OR cm.{nameof(ChannelMember.MemberId)} = @{nameof(member2Id)}) 
```

Comment:
I do not see the second part of the initial request: ((cm.MemberId = @member1Id OR cm.MemberId = @member2Id) **AND (c.CreatorId = @member1Id OR c.CreatorId = @member2Id))**

Reply:
done